app/queries/servers/channel.test.ts

```diff
@@ -0,0 +1,1552 @@
+import type ChannelMembershipModel from '@typings/database/models/servers/channel_membership';
+import type MyChannelModel from '@typings/database/models/servers/my_channel';
+import type MyChannelSettingsModel from '@typings/database/models/servers/my_channel_settings';
+
+jest.mock('./role');
+jest.mock('./system');
+jest.mock('./user');
+jest.mock('@utils/role');
+
+describe('prepareChannels', () => {
+    let operator: ServerDataOperator;
+
+    beforeEach(() => {
+        operator = {
+            handleChannel: jest.fn(),
+            handleChannelInfo: jest.fn(),
+            handleChannelMembership: jest.fn(),
+            handleMyChannel: jest.fn(),
+            handleMyChannelSettings: jest.fn(),
+        } as unknown as ServerDataOperator;
+    });
+
+    it('should prepare channels, channelInfos, channelMemberships, memberships, and myChannelSettings', () => {
+        const channels = [{id: 'channel1'}, {id: 'channel2'}] as Channel[];
+        const channelInfos = [{id: 'channel1'}, {id: 'channel2'}] as ChannelInfo[];
+        const channelMemberships = [{channel_id: 'channel1', user_id: 'user1'}, {channel_id: 'channel2', user_id: 'user2'}] as ChannelMembershipsExtended[];
+        const memberships = [{channel_id: 'channel1', user_id: 'user1'}, {channel_id: 'channel2', user_id: 'user2'}] as ChannelMembership[];
+
+        const channelRecords = Promise.resolve([]);
+        const channelInfoRecords = Promise.resolve([]);
+        const membershipRecords = Promise.resolve([]);
+        const myChannelRecords = Promise.resolve([]);
+        const myChannelSettingsRecords = Promise.resolve([]);
+
+        (operator.handleChannel as jest.Mock).mockReturnValue(channelRecords);
+        (operator.handleChannelInfo as jest.Mock).mockReturnValue(channelInfoRecords);
+        (operator.handleChannelMembership as jest.Mock).mockReturnValue(membershipRecords);
+        (operator.handleMyChannel as jest.Mock).mockReturnValue(myChannelRecords);
+        (operator.handleMyChannelSettings as jest.Mock).mockReturnValue(myChannelSettingsRecords);
+
+        const result = prepareChannels(operator, channels, channelInfos, channelMemberships, memberships, true);
+
+        expect(result).toEqual([channelRecords, channelInfoRecords, membershipRecords, myChannelRecords, myChannelSettingsRecords]);
+        expect(operator.handleChannel).toHaveBeenCalledWith({channels, prepareRecordsOnly: true});
+        expect(operator.handleChannelInfo).toHaveBeenCalledWith({channelInfos, prepareRecordsOnly: true});
+        expect(operator.handleChannelMembership).toHaveBeenCalledWith({channelMemberships, prepareRecordsOnly: true});
+        expect(operator.handleMyChannel).toHaveBeenCalledWith({channels, myChannels: memberships, prepareRecordsOnly: true, isCRTEnabled: true});
+        expect(operator.handleMyChannelSettings).toHaveBeenCalledWith({settings: memberships, prepareRecordsOnly: true});
```

Comment:
See all results are empty, the tests is used to pass the code coverage but is not actually testing the preparation of models.

I'm fine with that for now, but we should probably file another ticket to improve the test

Reply:
Should these tests be testing the preparation of the models, though? I don't think so. We should have tests on the operator to do that. These tests should only test the specific functionality of the query functions